atc/db/pipeline.go

```diff
@@ -653,6 +654,40 @@ func (p *pipeline) Rename(name string) error {
 	return err
 }
 
+func (p *pipeline) Move(destinationTeamName string) error {
+	destinationTeamID, err := psql.Select("id").
+		From("teams").
+		Where(sq.Eq{
+			"name": destinationTeamName,
+		}).
+		RunWith(p.conn).Query()
+
+	_, err = psql.Update("pipelines").
+		Set("team_id", destinationTeamID).
+		Where(sq.Eq{
+			"id": p.id,
+		}).
+		RunWith(p.conn).
+		Exec()
+
+	if err != nil {
+		return err
+	}
+
+	_, err = psql.Update("builds").
```

Comment:
I'm a little worried about this query because it will be updating all the builds within a pipeline, which can result in thousands or millions of rows being updated. Maybe it should be done in batches? Not too sure yet tho.. 🤔 I might want to do some bench marking to see how long it will take!

Reply:
Is it enough that we do a count check before making this update and abort with an error message if the count is more than 100 or something? That way we will know if this feature is in fact used by people with pipelines that have thousands of rows. If not atleast there will be some protection.

Reply:
also added the transaction logic.

Reply:
If we do a count then abort, that would mean users that have more builds than our limit that we set wouldn't be able to use this `move` command. I think most of Concourse users have pipelines that have more than 100 or even 1000 builds 🤔 so I was more thinking of doing query in batches of 1000 builds. Doing it in batches will make it so that it doesn't lock up all the build rows while it's trying to run the update.